packages/next/src/cli/next-lint.ts

```diff
@@ -196,6 +198,14 @@ const nextLint: CliCommand = async (argv) => {
   const distDir = join(baseDir, nextConfig.distDir)
   const defaultCacheLocation = join(distDir, 'cache', 'eslint/')
   const hasAppDir = !!nextConfig.experimental.appDir
+  const { pagesDir } = findPagesDir(baseDir, !!nextConfig.experimental.appDir)
+
+  await writeAppTypeDeclarations({
```

Comment:
We should probably use `verifyTypescriptSetup` here to ensure the same configs from build/dev are updated here as well x-ref: https://github.com/vercel/next.js/blob/bd9ad97c8e2f996a41ef6d69fd0445a837339bad/packages/next/src/server/dev/next-dev-server.ts#L794

Reply:
oh, seems I missed this, sure, will add it now